src/main/java/org/spectrumauctions/sats/mechanism/domains/ModelDomain.java

```diff
@@ -1,42 +1,66 @@
 package org.spectrumauctions.sats.mechanism.domains;
 
-import lombok.EqualsAndHashCode;
-import lombok.Getter;
+import java.math.BigDecimal;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.apache.commons.math3.stat.regression.SimpleRegression;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.marketdesignresearch.mechlib.core.Allocation;
+import org.marketdesignresearch.mechlib.core.BundleEntry;
 import org.marketdesignresearch.mechlib.core.Domain;
+import org.marketdesignresearch.mechlib.core.Good;
+import org.marketdesignresearch.mechlib.core.bidder.valuefunction.BundleValue;
+import org.marketdesignresearch.mechlib.core.price.LinearPrices;
+import org.marketdesignresearch.mechlib.core.price.Price;
+import org.marketdesignresearch.mechlib.core.price.Prices;
 import org.marketdesignresearch.mechlib.instrumentation.MipInstrumentation;
-import org.spectrumauctions.sats.core.model.GenericWorld;
+import org.spectrumauctions.sats.core.bidlang.BiddingLanguage;
 import org.spectrumauctions.sats.core.model.SATSBidder;
-import org.spectrumauctions.sats.core.model.SATSGood;
+import org.spectrumauctions.sats.core.model.UnsupportedBiddingLanguageException;
+import org.spectrumauctions.sats.core.util.random.JavaUtilRNGSupplier;
+import org.spectrumauctions.sats.core.util.random.RNGSupplier;
 import org.spectrumauctions.sats.opt.model.ModelMIP;
 
-import java.util.List;
+import lombok.AccessLevel;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
 
 @EqualsAndHashCode
-public abstract class ModelDomain implements Domain {
+public abstract class ModelDomain<T extends SATSBidder> implements Domain {
+	
+	private static final Logger logger = LogManager.getLogger(ModelDomain.class);
 
     @Getter
-    private List<? extends SATSBidder> bidders;
-
-    @Getter
-    private List<? extends SATSGood> goods;
+    private List<T> bidders;
 
     private transient Allocation efficientAllocation;
 
     protected abstract ModelMIP getMIP();
-
-    public ModelDomain(List<? extends SATSBidder> bidders) {
+    
+    private boolean generic;
+    
+    @Setter @Getter(value = AccessLevel.PROTECTED)
+    private int priceGenerationBidsPerBidder = 100;
+    @Setter
+    private int priceGenerationNumberOfWorldSamples = 100;
+    @Setter
+    private double priceGenerationFraction = 0.01;
+    @Setter
+    private long priceGenerationSeed = System.currentTimeMillis();
+    
+    
+    public ModelDomain(List<T> bidders) {
         this(bidders, false);
     }
 
-    public ModelDomain(List<? extends SATSBidder> bidders, boolean generic) {
+    public ModelDomain(List<T> bidders, boolean generic) {
```

Comment:
Maybe I overlooked something, but you can simplify the constructors here since there's no distinction on this level between generic and non-generic, no?

Reply:
You are right. I removed this constructor